### DISCUSSION TITLE
Facing an issue with <ClientOnly>

### DISCUSSION THREAD

**Discussion by Croodo** [nuxt/nuxt#18666] [Questions]
Everything works fine when navigating through **_NuxtLink_** but on page load devtools through this error: 
`Uncaught (in promise) TypeError: Cannot read properties of null (reading 'list')`

Without `<ClientOnly>` it works perfectly. 
With `<ClientOnly>` content is rendered but gives some error when doing some functional stuff.

MyComponent.vue
```
<script setup>
const { data: list } = await useFetch("/api/endpoint");</script>
<template>
 <div
    class="block"
    v-for="(video, index) in list"
    :key="index"
  >
{{ video }}
</div>
</template>
```

page.vue
```
<template>
<div>
<ClientOnly><MyComponent /></ClientOnly>
  </div>
</template>
```

**Comment by Baroshem** [comment]
My bet would be that you are trying to destructure data from useFetch (that works on the server side and then hydrates the browser) and then you want to put it into the browser only component. 

It works on NuxtLink as it is browser-only navigation. The hard reload of the page is coming from the server.

I would recommend to avoid mixing the client-only component with server logic (from useFetch or useAsyncData)
